### DISCUSSION TITLE
Configuring Lemmy Without Pict-rs and/or Image Caching?

### DISCUSSION THREAD

**Discussion by johnriggs** [ubergeek77/Lemmy-Easy-Deploy#71] [Q&A]
First off please let me say that I am very grateful for the work you've done on this tool! It's really simplified standing up a Lemmy instance for those of us who are very new to this sort of thing.

I was running a small instance until yesterday. Due to the despicable attacks on lemmy.world - and me not being able to answer whether or not illegal contact had federated to my server - I chose to shut it down until I can be sure that sort of attack won't be a threat to my instance. 

But I still want to participate in the Fediverse and help share the load; I think at this point I just don't want to support local image hosting or caching of federated images. 

Is there a clear way of setting up Lemmy (for us noobies) that addresses these concerns? 

Thanks in advance for any and all answers and insights!

**Comment by ubergeek77** [comment]
Thank you for the kind words!

Honestly, I think the future of this recent issue will end up being a config option in Lemmy to turn off these things. As this project is simply an automation layer that takes the components/services Lemmy puts out and deploys them, I don't have much control over the lifecycle of images.

If/when they do add that config option, no changes will be required by this script, as that file will just be editable as-is.

As important as this issue is, I'm not interested in maintaining support for one deployment of Lemmy with pict-rs, and one without. I also don't think that's the correct approach to this issue.

To simplify my workload, and not introduce a complicated permutation of "flavors" of Lemmy that can be deployed, I keep this deployment as aligned to the official Lemmy Ansible deployment as possible. If they make a change, I will adopt it, but I avoid adding or removing services/features based on preference. So since they deploy pict-rs, so do I.

For now, if you really want to cut pictrs out of your deployment entirely, you are welcome to modify this script to suit your needs, but I won't be removing pict-rs from this project altogether.

I can't provide support for these kinds of deployments, but I can at least give you some hints.

Check out the Advanced Configuration page:

https://github.com/ubergeek77/Lemmy-Easy-Deploy/blob/main/ADVANCED_CONFIGURATION.md

You will likely want to make a copy of the default docker-compose template, remove the pictrs service, then find the part of my script that checks if a service is alive. Remove pictrs from that list, so that the deployment doesn't fail if pictrs is missing.

**Comment by nikidubois** [comment]
Thank you for your feedback. I understand your standpoint on things. However, I too hope that this feature will be available soon, and preferably via an official Lemmy setting.

However, things currently are that it is not available via an official setting, and there are multiple problems with pictrs.
1. super growth of space required to run Lemmy (I currently need more than 50 gb webspace, that used to be 10 max)
2. illegal federated content via pictures making an admin responsible
3. possibly being hacked
and I might not be aware about other things.

I see this install as an easy way to run Lemmy. This makes most of your users not that good in running all things individually (otherwise we might just run directly via ansible).

Removing pictrs is almost not possible for me. I asked for assistance via Lemmy, and got the response "remove it from your docker compose".

I know a thing or 2, but I'm not "good" at these things. Again, if I was, I would not use your script.

So, at the moment I have no idea on how to do this - I can't cut it out the docker compose in the Live folder, because that gets overwritten. This is about as far as my knowledge goes.

I absolutely follow your writing, but I do ask you, because of the current situation and a solution not being ready yet, to review your standpoint about this.
A setting via config.env would be a perfectly good solution. (No idea how much work this would be for you...)

Thanks for listening to my ted talk.

**Comment by ubergeek77** [comment]
If I do allow the user to remove pictrs, it would be with the following caveats:

- Thumbnails will not work at all. Your instance will feel very barren. Links to websites will show a blank icon instead of a thumbnail. That's just how Lemmy works and I don't have the ability to create a custom solution for this.
- Lemmy is ***designed*** to run with pictrs. It is a mandatory service. It will work generally fine without it, but the Lemmy project makes assumptions about it being available, so weird bugs will occur. In fact, if you remove pictrs, I can guarantee you your UI will stop working if your instance has an icon configured, because the Lemmy UI simply crashes if it can't download the icon. If something breaks due to pictrs being missing, such as a crash if Lemmy tries to retrieve a thumbnail, I cannot provide support, as I am not a developer of Lemmy. The removal of pictrs will be considered by this project to be an unsupported "advanced" feature.

Removing pictrs is **a hacky solution**. It's a kneejerk reaction to a bigger problem, and unfortunately it's balancing on the line of what I can feasibly keep in scope for ***this*** project. I will think about it though, but right now I'm trying to balance my day job and a full move to a different house, so I'm at really limited capacity to do these sorts of things at the moment. I'll try to get to this as soon as I can, but no promises on when 🙏

**Reply by nikidubois** [reply]
Very valid response. I had no idea that Lemmy would be buggy and/or things like that would happen. What I understood from Lemmy comments was that the instance would just go and download the images from the source and that pictrs would just act as a caching mechanism to not constantly download all images.

You are right, I do want my thumbnails. I do want my banner and icon to work. Those are not things I wish to give up on.

Good luck with your move and no worries. Let's just hope that there will be an official fix soon.

Again, thanks for your great work!
